tests/bgp/test_route_aggregation.py

```diff
@@ -0,0 +1,195 @@
+    }
+
+    logger.info("DUT Config After Setup: {}".format(duthost.shell("show run bgp",
+                module_ignore_errors=True)['stdout']))
+
+    yield setup_info
+
+    # restore config to original state
+    config_reload(duthost, wait=60)
+
+    # verify sessions are established
+    bgp_facts = duthost.bgp_facts(instance_id=asic_index)['ansible_facts']
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'] == tor1:
+            logger.debug(v['description'])
+            assert v['state'] == 'established'
+
+
+def test_ebgp_route_aggregation(setup):
+    # aggregate directly connected routes
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "address-family ipv4 unicast" \
+        -c "aggregate-address {} summary-only" -c "address-family ipv6 unicast" \
+        -c "aggregate-address {} summary-only"'.format(setup['namespace'], setup['dut_asn'], NEI_IPv4_AGG_ROUTE,
+                                                       NEI_IPv6_AGG_ROUTE)
+    setup['duthost'].shell(cmd, module_ignore_errors=True)
+
+    cmd = 'vtysh -n {} -c "clear bgp * soft"'.format(setup['namespace'])
+    setup['duthost'].shell(cmd, module_ignore_errors=True)
+    time.sleep(establish_bgp_session_time)
+
+    logger.info("DUT Config After Aggregation: {}".format(setup['duthost'].shell("show run bgp",
+                module_ignore_errors=True)['stdout']))
+
+    cmd = "show ip bgp neighbors {} received-routes".format(setup['dut_ip_v4'])
+    nei_show_neigh_v4 = setup['neighhost'].shell(cmd, module_ignore_errors=True)['stdout']
+    cmd = "show ipv6 bgp neighbors {} received-routes".format(setup['dut_ip_v6'])
+    nei_show_neigh_v6 = setup['neighhost'].shell(cmd, module_ignore_errors=True)['stdout']
+    logger.info("BGP Neighbors IPv4: {}\n\nBGP Neighbors IPv6: {}".format(nei_show_neigh_v4, nei_show_neigh_v6))
+    ipv4_agg_route_present = False
+    ipv6_agg_route_present = False
+    if NEI_IPv4_AGG_ROUTE in nei_show_neigh_v4:
+        ipv4_agg_route_present = True
+    if NEI_IPv6_AGG_ROUTE in nei_show_neigh_v6:
+        ipv6_agg_route_present = True
+    assert ipv4_agg_route_present is True
+    assert ipv6_agg_route_present is True
+
+    # verify individual routes are not being received
+    ipv4_route_present = False
+    ipv6_route_present = False
+    if nei_ipv4_route_1 in nei_show_neigh_v4:
+        ipv4_route_present = True
+    if nei_ipv6_route_1 in nei_show_neigh_v6:
+        ipv6_route_present = True
+    if nei_ipv4_route_2 in nei_show_neigh_v4:
+        ipv4_route_present = True
+    if nei_ipv6_route_2 in nei_show_neigh_v6:
+        ipv6_route_present = True
+    assert ipv4_route_present is False
+    assert ipv6_route_present is False
+
+    # aggregate directly connected routes with as-set
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "address-family ipv4 unicast" \
+        -c "no aggregate-address {} summary-only" -c "aggregate-address {} as-set summary-only" \
```

Comment:
as-set verification is not done in the show output.

Reply:
Can you please point out where this comment is handled.

Reply:
@kishorekunal01 This is checked starting on line 67.